snuba/query/data_source/join.py

```diff
@@ -29,6 +39,18 @@ class JoinClass(Enum):
     N_2_N = "N-N"
 
 
+class JoinRelationship(NamedTuple):
+    """
+    Represent the join relationship between an entity and another entity.
+    """
+
+    rhs_entity: EntityKey
+    join_class: JoinClass
+    join_type: JoinType
+    join_modifier: JoinModifier
+    columns: Sequence[Tuple[str, str]]
```

Comment:
One question that came up while I was adding a concept of column semantic equivalence between columns of different entities:
If we have two relationship with different names between entities A and B, would it be correct to say that such two relationship may have different columns field ?

Reply:
Yes definitely, each relationship is completely independent. I think it would be valid to have two identical relationships with different names. I'm not sure what the purpose of that would be, but I don't think it would be incorrect.